### PR TITLE
[AAP-6818] Configure controller API access

### PR DESCRIPTION
**Pull request by bzwei**
Take controller base url and authentication token from CLI options or through websocket in worker mode.

Resolves AAP-6818

Includes all commits from #243  #253. This concludes the backend support of run_job_template action using a non-SSL connection.
Future work will enable SSL.

### CODE REVIEW

File: ansible_rulebook/cli.py [hunk 1]
```diff
@@ -167,6 +179,14 @@ def main(args: List[str] = None) -> int:
         print("Error: inventory is required")
```
Comment (Alex-Izquierdo) [inline ansible_rulebook/cli.py:179]: This is not related with this PR but I'm looking this line and thinking that now inventory should not be mandatory with run_job_template action.

**Review by Alex-Izquierdo — CHANGES_REQUESTED** [review]
Documentation should indicate that these flags must be defined to use this action

**Comment by Alex-Izquierdo** [comment]
Is there some specification about how must be provided the credentials? I wonder if we should support environment variables as well.

**Comment by Alex-Izquierdo** [comment]
> Documentation should indicate that these flags must be defined to use this action

Leave it to me please. I'm working on a big refactor of the docs and the rebase would be a pain.

File: ansible_rulebook/cli.py (hunk 1 above)
Comment (bzwei) [reply ansible_rulebook/cli.py:179]: agree

**Comment by bzwei** [comment]
@Alex-Izquierdo Is there any code change needed for this PR?

**Review by mkanoor — APPROVED** [review]

**Pull request merged**